src/relations/db.py

```diff
@@ -136,6 +138,60 @@ def __init__(self, charm: CharmBase, admin: bool = False):
     def _depart_flag(self, relation):
         return f"{self.relation_name}_{relation.id}_departing"
 
+    def _check_extensions(self, extensions: List) -> bool:
+        """Checks if requested extensions are enabled."""
+        for extension in extensions:
+            extension_name = extension.split(":")[0]
+            if not self.charm.backend.database.is_postgresql_plugin_enabled(extension_name):
+                return False
+        return True
+
+    def _block_on_extensions(self, relation, remote_app_databag: Dict) -> bool:
+        """Verifies that extensions are enabled or blocks the charm."""
+        if "extensions" in remote_app_databag:
+            self.update_databags(
+                relation,
+                {
+                    "extensions": remote_app_databag["extensions"],
+                },
+            )
+            extensions = remote_app_databag["extensions"].split(",")
+            if not self._check_extensions(extensions):
+                # Do not allow apps requesting extensions to be installed.
+                logger.error(
+                    f"ERROR - `extensions` ({', '.join(extensions)}) cannot be requested through relations"
+                    " - Please enable extensions through `juju config` and add the relation again."
```

Comment:
In this scenario:

> juju deploy postgresql, pgbouncer, indico
> juju relate postgresql<>pgbouncer<>indico # pgbouncer will be blocked due to the requested extension(s)
> juju config postgresql plugin_pg_trgm_enable=True

Q: can PGB be notified about config changes on PG? Sure, PGB will see new relation data on the next update_status, but it is too long...

Reply:
At the moment the check will happen only on joining the relation. The intended user behaviour is to reconfigure the Postgresql charm and rerelate with Pgbouncer. I can add the check to update_status as well, but I don't think we should do more, as this is an escape hatch for users to be able to migrate away from the legacy relation and not something we intend to support indefinitely.

Reply:
As Dragomir explained, I think is better to tell the user that the application charm needs to be relate again to PgBouncer after the extensions are enabled.

Currently, we don't have any field in the `postgresql_client` relation to tell that a plugin was enabled. On https://github.com/canonical/charm-relation-interfaces/pull/74 I started to create a field, but then Mykola told me that the `DataRequires` library should actively (through `is_postgresql_plugin_enabled `) check whether a plugin was enabled.